File: opspilot-enterprise/packages/shared-types/src/topology.ts
export interface TopologyNode {
  id: string;
  name: string;
  type: string;
  status?: string;
  metadata?: Record<string, unknown>;
}

export interface TopologyEdge {
  id: string;
  source: string;
  target: string;
  relation: string;
  metadata?: Record<string, unknown>;
}

export interface TopologyGraph {
  graph_id: string;
  connection_id: string;
  generated_at: string;
  nodes: TopologyNode[];
  edges: TopologyEdge[];
  metadata?: Record<string, unknown>;
}

File: packages/shared-schema/src/opspilot_schema/topology.py
from __future__ import annotations

from pydantic import BaseModel


class TopologyNode(BaseModel):
    id: str
    name: str
    type: str
    status: str | None = None
    metadata: dict = {}


class TopologyEdge(BaseModel):
    id: str
    source: str
    target: str
    relation: str
    metadata: dict = {}


class TopologyGraph(BaseModel):
    graph_id: str
    connection_id: str
    generated_at: str
    nodes: list[TopologyNode] = []
    edges: list[TopologyEdge] = []
    metadata: dict = {}
